src/GraphQL/ResolveFieldContext/ResolveFieldContextExtensions.cs

```diff
@@ -120,5 +120,92 @@ public static async Task<TResult> TryAsyncResolve<TResult>(this IResolveFieldCon
                 }
             }
         }
+
+        private static readonly char[] _separators = new char[] { '.' };
+
+        /// <summary>
+        /// Thread safe method to get value by path (key1.key2.keyN) from extensions dictionary.
+        /// </summary>
+        /// <param name="context">Context with extensions response map.</param>
+        /// <param name="path">Path to value in key1.key2.keyN format.</param>
+        /// <returns>Value, if any exists on the specified path, otherwise <c>null</c>.</returns>
+        public static object GetExtension(this IResolveFieldContext context, string path)
+        {
+            if (context == null)
+                throw new ArgumentNullException(nameof(context));
+
+            if (context.Extensions == null || context.Extensions.Count == 0)
+                return null;
+
```

Comment:
why not make extensions a concurrent dictionary?

Reply:
Because then it would be required to do concurrent dictionary at each level of nesting which is impossible. It would be strange to provide concurrent dictionary only at the first level. In fact, the user can set his own dictionary as value (not concurrent) and then the thread safety behavior will be different from level to level. So I decided to use regular dictionaries and 2 thread safe extension methods.